Keen/ContractResolvers/SnakeCaseContractResolver.cs

```diff
@@ -0,0 +1,55 @@
+﻿namespace Keen.Core.ContractResolvers
+{
+    using Newtonsoft.Json.Serialization;
+
+    /* 
+     * Found on GithubGist https://gist.github.com/crallen/9238178
+     * 
+     * Credit to Chris Allen https://github.com/crallen 
+     */
+    public class SnakeCaseContractResolver : DefaultContractResolver
```

Comment:
I've used the Contract Resolver rather than Json.NET because all the attributes defined on all the existing models are using the JsonPropertyAttribute from Newtonsoft.Json. Which will only be respected when using the `JsonConvert.SerializeObject()` and `JsonConvert.DeserializeObject()` functions.

The `SnakeCaseContractResolver` doesnt handle `camelCase`, `PascalCase` or `kebab-case` but custom resolvers may be defined for those. The resolver is defined at serialization/deserialization time by passing it as an option to JsonConvert like so, rather than in a global setting:
```
JsonConvert.DeserializeObject<DatasetDefinition>(responseString, new JsonSerializerSettings { ContractResolver = new SnakeCaseContractResolver()});
```

This, and indeed any ContractResolver extending Newtonsoft.Json's `DefaultContractResolver` will respect all JsonPropertyAttributes